src/plugins/workspace/server/plugin.ts

```diff
@@ -156,8 +157,13 @@ export class WorkspacePlugin implements Plugin<WorkspacePluginSetup, WorkspacePl
       workspaces: {
         enabled: true,
         permissionEnabled: isPermissionControlEnabled,
+        isDashboardAdmin: true,
       },
     }));
+    core.capabilities.registerSwitcher((request) => {
+      const isDashboardAdmin = getWorkspaceState(request).isDashboardAdmin !== false;
+      return { workspaces: { isDashboardAdmin } };
```

Comment:
Better to add some comment here to elaborate.

And, should we add unit test to test 3 scenario? : true / false / undefined.

Reply:
Sure, updated.